tests/test_inputmedia.py

```diff
@@ -0,0 +1,136 @@
+
+    def test_with_video(self, video):
+        # fixture found in test_video
+        imv = InputMediaVideo(video, caption="test 3")
+        assert imv.type == self.type
+        assert imv.media == video.file_id
+        assert imv.width == video.width
+        assert imv.height == video.height
+        assert imv.duration == video.duration
+        assert imv.caption == "test 3"
+
+    def test_error_with_file(self, video_file):
+        # fixture found in test_video
+        with pytest.raises(ValueError, match="only support file_id or url as a valid media"):
+            InputMediaPhoto(video_file)
+
+
+class TestInputMediaPhoto(object):
+    type = "photo"
+    media = "NOTAREALFILEID"
+    caption = "My Caption"
+
+    def test_expected_values(self, input_media_photo):
+        assert input_media_photo.type == self.type
+        assert input_media_photo.media == self.media
+        assert input_media_photo.caption == self.caption
+
+    def test_to_dict(self, input_media_photo):
+        input_media_photo_dict = input_media_photo.to_dict()
+        assert input_media_photo_dict['type'] == input_media_photo.type
+        assert input_media_photo_dict['media'] == input_media_photo.media
+        assert input_media_photo_dict['caption'] == input_media_photo.caption
+
+    def test_with_photo(self, photo):
+        # fixture found in test_photo
+        imp = InputMediaPhoto(photo, caption="test 2")
+        assert imp.type == self.type
+        assert imp.media == photo.file_id
+        assert imp.caption == "test 2"
+
+    def test_error_with_file(self, photo_file):
+        # fixture found in test_photo
+        with pytest.raises(ValueError, match="only support file_id or url as a valid media"):
+            InputMediaPhoto(photo_file)
+
+
+@pytest.fixture(scope='function')
+def media_group(photo, thumb):
+    return [InputMediaPhoto(photo), InputMediaPhoto(thumb)]
+
+
+class TestSendMediaGroup(object):
+    @flaky(3, 1)
+    @pytest.mark.timeout(10)
+    def test_send_media_group_photo(self, bot, chat_id, media_group):
+        messages = bot.send_media_group(chat_id, media_group)
+        assert isinstance(messages, list)
+        assert len(messages) == 2
+        assert all([isinstance(mes, Message) for mes in messages])
+        assert all([mes.media_group_id == messages[0].media_group_id for mes in messages])
+
+    @flaky(3, 1)
+    @pytest.mark.timeout(10)
+    def test_send_media_group_all_args(self, bot, chat_id, media_group):
+        m1 = bot.send_message(chat_id, text="test")
+        messages = bot.send_media_group(chat_id, media_group, disable_notification=True,
+                                        reply_to_message_id=m1.message_id)
+        assert isinstance(messages, list)
+        assert len(messages) == 2
+        assert all([isinstance(mes, Message) for mes in messages])
+        assert all([mes.media_group_id == messages[0].media_group_id for mes in messages])
```

Comment:
Maybe a TODO: about needing to test with new files.

Reply:
yup